package dsa.BinarySearch;
class Solution {
    public int findRotationPoint(int[] arr) {
        int left = 0;
        int right = arr.length - 1;

        while (left < right) {
            int mid = left + (right - left) / 2;

            if (arr[mid] > arr[right]) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return left; // index of the smallest element (rotation point)
    }

    public static void main(String[] args) {
        Solution obj = new Solution();

        int[] arr = {4, 5, 6, 7, 0, 1, 2};

        int index = obj.findRotationPoint(arr);

        System.out.println("Rotation Point Index: " + index);
        System.out.println("Smallest Element: " + arr[index]);
    }
}